server/embed/etcd.go

```diff
@@ -770,3 +794,24 @@ func parseCompactionRetention(mode, retention string) (ret time.Duration, err er
 	}
 	return ret, nil
 }
+
+func (e *Etcd) setupTracing() (tracesdk.SpanExporter, error) {
+	exporter, err := otlp.NewExporter(e.tctx,
+		otlpgrpc.NewDriver(
+			otlpgrpc.WithEndpoint(e.cfg.ExperimentalTracingAddress),
+			otlpgrpc.WithInsecure(),
+		))
+	if err != nil {
+		return nil, err
+	}
+	res := resource.NewWithAttributes(
```

Comment:
is `ExperimentalTracingServiceName` mandatory? can it be empty?

Reply:
It can be empty, it has a default value of `"etcd"`. But I would say we should recommend to users to make use of it, for the reason detailed in https://github.com/etcd-io/etcd/pull/12919#discussion_r627103132.

The only thing you need to pass to enable tracing with OpenTelemetry is the enable flag `etcd --experimental-enable-tracing`.

Reply:
Setting service name differently for each instance of a horizontally scaled service goes against the Otel conventions: https://github.com/open-telemetry/opentelemetry-specification/blob/e440dd50a16f4ff4f5ff47e420a49129127f7a37/specification/resource/semantic_conventions/README.md#service.  We should use other attributes (e.g. service.instance.id) to differentiate instances of etcd from each other.  Service.name should be a constant.

Reply:
Interesting approach, does this get translated into the tag in jaeger UI?

Reply:
Yes.  Attributes are just key-value pairs, so you can add whatever attributes you want to, and they will show up in the Jaeger UI.  In general, we should try to use semantic conventions whenever possible, but if we have our own attribute that we don't think maps well to existing conventions (e.g. etcd replica number), we can just add our own here.

Reply:
Added ability of setting those now, but haven't found a way to filter by them yet in the jaeger UI. As it shows up under special filed not under tags for me.

But good suggestion!